common/djangoapps/student/views.py

```diff
@@ -1670,16 +1670,14 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
-    # Boolean of whether a 3rd party auth provider and credentials were provided in
-    # the API so the newly created account can link with the 3rd party account.
-    #
-    # Note: this is orthogonal to the 3rd party authentication pipeline that occurs
-    # when the account is created via the browser and redirect URLs.
-    should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
-
-    if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
+    if third_party_auth.is_enabled() and (pipeline.running(request) or 'provider' in params):
         params["password"] = pipeline.make_random_password()
 
+    # in case user is registering via third party and pipeline has expired, show appropriate
+    # error message
+    if 'is_third_party_request' in params and not pipeline.running(request):
+        raise ValidationError({'session_expired': ["Your session with third party has been expired"]})
```

Comment:
@noraiz-anwar @nasthagiri I don't think we need to point out that the third party provider is a third party provider, so if we go with that suggestion, I'd prefer just ``Registration using {provider_name} has timed out``. **However:** I think that message is less helpful to the user than "We couldn't create your account. Try again in a few minutes". Just telling her that registration timed out may leave her wondering if her account was created, and what she should do—should she try again right away? If she does that and her account was already created, will she end up with two accounts? The end result that the user is concerned about is that her account wasn't created, and the action we want her to take is to wait a few minutes and try again. She won't necessarily care that the reason her account wasn't created was that registration timed out; she can't do anything with that information.

If we really want to include information about the third party provider, I would suggest this:

``We couldn't create your account.``
``Registration using {provider_name} timed out. Wait a few minutes and try again.``

Or this:

``A problem occurred when we tried to create your account.``
``{provider_name} did not respond. Wait a few minutes and try again.``